src/system/connectivity/wifi.cpp

```diff
@@ -15,13 +40,29 @@ static WiFiManager wm;
  * @param wifi_password  [Optional] WiFi Password as a const char string.
  * @return wl_status_t   WiFi status code.
  */
-wl_status_t WifiInitialize (
+    wl_status_t WiFiConnection::WifiInitialize (
     const char* wifi_ssid,
     const char* wifi_password
     )
 {
     logger << LOG_INFO << "Initializing WiFi provider!" << EndLine;
 
+    WiFiManagerParameter templateID("template_id", "TEMPLATE_ID", "template_id",20);
+    WiFiManagerParameter deviceName("device_name", "DEVICE_NAME", "Device_Name", 20);
+    WiFiManagerParameter authToken("auth_token", "AUTH_TOKEN", "auth_token", 20);
+
+    wm.addParameter(&templateID);
+    wm.addParameter(&deviceName);
+    wm.addParameter(&authToken);
+
+    wm.setSaveParamsCallback(saveParamCallback);
+
+    wm.setTitle("Irrigation System");
+    std::vector<const char *> menu = {"wifi","info","param","sep","restart","exit"};
```

Comment:
Why the usage of std::vector?
Adding STL stuff is not increasing the size of the firmware?

Reply:
Also, this could be a const array.